caching-service/src/main/java/org/zowe/apiml/caching/service/infinispan/storage/InfinispanStorage.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.caching.service.infinispan.storage;
+
+import lombok.extern.slf4j.Slf4j;
+import org.infinispan.Cache;
+import org.zowe.apiml.caching.model.KeyValue;
+import org.zowe.apiml.caching.service.Messages;
+import org.zowe.apiml.caching.service.Storage;
+import org.zowe.apiml.caching.service.StorageException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
+public class InfinispanStorage implements Storage {
+
+
+    private final Cache<String, Map<String, KeyValue>> cache;
+
+    public InfinispanStorage(Cache<String, Map<String, KeyValue>> cache) {
+        this.cache = cache;
+    }
+
+    @Override
+    public KeyValue create(String serviceId, KeyValue toCreate) {
+        log.info("Writing record: {}|{}|{}", serviceId, toCreate.getKey(), toCreate.getValue());
+
+        Map<String, KeyValue> serviceCache = cache.computeIfAbsent(serviceId, k -> new HashMap<>());
+
+        if (serviceCache.containsKey(toCreate.getKey())) {
+            throw new StorageException(Messages.DUPLICATE_KEY.getKey(), Messages.DUPLICATE_KEY.getStatus(), toCreate.getKey());
+        }
+        KeyValue entry = serviceCache.put(toCreate.getKey(), toCreate);
+        cache.put(serviceId, serviceCache);
+        return entry;
+    }
+
+    @Override
+    public KeyValue read(String serviceId, String key) {
+        Map<String, KeyValue> serviceCache = cache.get(serviceId);
```

Comment:
Could you add log.info also to the other methods?

Reply:
logs added